solr/core/src/java/org/apache/solr/handler/admin/api/CoreReplicationAPI.java

```diff
@@ -46,12 +50,24 @@ public CoreReplicationAPI(SolrCore solrCore, SolrQueryRequest req, SolrQueryResp
 
   @GET
   @Path("/indexversion")
-  @Produces({"application/json", "application/xml", BINARY_CONTENT_TYPE_V2})
+  @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML, BINARY_CONTENT_TYPE_V2})
   @PermissionName(CORE_READ_PERM)
   public IndexVersionResponse fetchIndexVersion() throws IOException {
     return doFetchIndexVersion();
   }
 
+  @GET
+  @Path("/files/generation/{gen}")
```

Comment:
[Q] For a lot of these v2 APIs, I've been working from a [spreadsheet](https://docs.google.com/spreadsheets/d/1HAoBBFPpSiT8mJmgNZKkZAPwfCfPvlc08m5jz3fQBpA) that describes what the new endpoints should be.

In the spreadsheet, the suggested v2 API was just `GET /cores/coreName/replication/files`.  (I think at the time I was assuming "generation" would be a required query parameter, rather than a path-param.  Though it's possible I didn't even know about it at that point.)

Not that we have to stick to that necessarily - if you think it makes sense to keep "generation" in the path, it's def an option.

Any thoughts?

Reply:
This seems reasonable to me, though it might be undermined if we decide to set a default "generation" value as we discussed elsewhere on this PR.  Let's consolidate discussion of this [here](https://github.com/apache/solr/pull/1704#discussion_r1238667428)